set-web/src/main/java/eu/europeana/set/web/service/controller/jsonld/WebUserSetRest.java

```diff
@@ -473,18 +439,15 @@ private void validateUpdateItemsByProfile(UserSet storedUserSet, UserSet updateU
 	 * @return response entity that comprises response body, headers and status code
 	 * @throws HttpException
 	 */
-	protected ResponseEntity<String> isItemInUserSet(HttpServletRequest request, String wsKey, 
-			String identifier, String datasetId, String localId, String userToken, 
+	protected ResponseEntity<String> isItemInUserSet(HttpServletRequest request,  
+			String identifier, String datasetId, String localId,  
 			String action) throws HttpException {
 
 		try {
 			// check user credentials, if invalid respond with HTTP 401,
 			//  or if unauthorized respond with HTTP 403
-			// check client access (a valid "wskey" must be provided)
-			validateApiKey(wsKey);
-
-			// authorize user
-			getAuthorizationService().authorizeUser(userToken, wsKey, identifier, Operations.RETRIEVE);
+		    // verify access rights
+			verifyReadAccess(request);
```

Comment:
this method needs to be invoked in the calling method

Reply:
verifyReadAccess invoked as suggested